app/javascript/chat/ChatChannelSettings/ChatChannelSettings.jsx

```diff
@@ -308,14 +308,20 @@ export default class ChatChannelSettings extends Component {
     const { currentMembership } = this.state;
     if (actionStatus) {
       const response = await leaveChatChannelMembership(currentMembership.id);
+      const { message } = response;
       if (response.success) {
+        this.setState({
+          successMessages: message,
+          errorMessages: null,
```

Comment:
**nitpick (non-blocking):** Should this maybe be `[]` instead? Personally I'm in favor of always returning the same datatype if possible, Maybe @nickytonline can weigh in on this too.

Reply:
It appears to be set to `null` whenever it's reset, but I agree it makes sense to keep the same type so that you don't need to make checks to see if it's `null` before doing an array operation.

Having said that, I looked through this whole component, and the `errorMessages` only seem to ever be getting set in component state but never used. Maybe I'm missing something. I noticed the same for the `successMessages`. I'm going to pull this locally.

Reply:
Alright, I pulled this locally and the type for `errorMessages` is a string, not an array, so `null` probably makes sense, but once again, having looked at the code and running it locally, I see `successMessages` and `errorMessages` get set in component state, but never actually used.

The only usage I see is when the response message gets destructured which has nothing to do with the component state. 🤔

Reply:
So I did another test where I removed `errorMessages` and `successMessages` from the component's code  and the application still works fine. If there is an an error, the error message is displayed in the Snack bar.

![image](https://user-images.githubusercontent.com/833231/101669146-8e41b480-3a1f-11eb-86a2-5c386a1c0313.png)

So although not the original intention of this PR, but from what I can tell and see, `errorMessages` and `successMessages` are never used in the code aside from being set in component state.

Reply:
Ah, my bad, the name suggested an array and I didn't get around to check out the code.